x/bsonx/bsoncore/value_test.go

```diff
@@ -677,163 +677,262 @@ func TestValue(t *testing.T) {
+			val: Value{
+				Type: TypeJavaScript,
+				Data: AppendJavaScript(nil, "code"),
+			},
+			want: `{"$code":"code"}`,
+		},
 
-		{"n>0, TypeJavaScript", 15, Value{
-			Type: TypeJavaScript, Data: AppendJavaScript(nil, "code")}, `{"$code":"code"`},
+		{
+			description: "TypeSymbol",
+			val: Value{
+				Type: TypeSymbol,
+				Data: AppendSymbol(nil, "symbol"),
+			},
+			want: `{"$symbol":"symbol"}`,
+		},
 
-		{"n>0, TypeSymbol", 10, Value{
-			Type: TypeSymbol, Data: AppendSymbol(nil, "symbol")}, `{"$symbol"`},
+		{
+			description: "TypeCodeWithScope",
+			val: Value{
+				Type: TypeCodeWithScope,
+				Data: AppendCodeWithScope(nil, "code",
+					BuildDocument(nil, AppendStringElement(nil, "key", "value")),
+				),
+			},
+			want: `{"$code":code,"$scope":{"key": "value"}}`,
+		},
 
-		{"n>0, TypeCodeWithScope", 10, Value{
-			Type: TypeCodeWithScope, Data: AppendCodeWithScope(nil, "code", BuildDocument(nil,
-				AppendStringElement(nil, "key", "value")))}, `{"$code":c`},
+		{
+			description: "TypeInt32",
+			val: Value{
+				Type: TypeInt32,
+				Data: AppendInt32(nil, 123),
+			},
+			want: `{"$numberInt":"123"}`,
+		},
 
-		{"n>0, TypeInt32", 10, Value{
-			Type: TypeInt32, Data: AppendInt32(nil, 123)}, `{"$numberI`},
+		{
+			description: "TypeTimestamp",
+			val: Value{
+				Type: TypeTimestamp,
+				Data: AppendTimestamp(nil, 123, 456),
+			},
+			want: `{"$timestamp":{"t":123,"i":456}}`,
+		},
 
-		{"n>0, TypeTimestamp", 10, Value{
-			Type: TypeTimestamp, Data: AppendTimestamp(nil, 123, 456)}, `{"$timesta`},
+		{
+			description: "TypeInt64",
+			val: Value{
+				Type: TypeInt64,
+				Data: AppendInt64(nil, 1234567890),
+			},
+			want: `{"$numberLong":"1234567890"}`,
+		},
 
-		{"n>0, TypeInt64", 10, Value{
-			Type: TypeInt64, Data: AppendInt64(nil, 1234567890)}, `{"$numberL`},
+		{
+			description: "TypeDecimal128",
+			val: Value{
+				Type: TypeDecimal128,
+				Data: AppendDecimal128(nil, 0x3040000000000000, 0x0000000000000000),
+			},
+			want: `{"$numberDecimal":"0"}`,
+		},
 
-		{"n>0, TypeDecimal128", 10, Value{
-			Type: TypeDecimal128, Data: AppendDecimal128(nil, 0x3040000000000000, 0x0000000000000000)}, `{"$numberD`},
+		{
+			description: "TypeMinKey",
+			val: Value{
+				Type: TypeMinKey,
+			},
+			want: `{"$minKey":1}`,
+		},
 
-		{"n>0, TypeMinKey", 10, Value{
-			Type: TypeMinKey}, `{"$minKey"`},
+		{
+			description: "TypeMaxKey",
+			val: Value{
+				Type: TypeMaxKey,
+			},
+			want: `{"$maxKey":1}`,
+		},
+	}
 
-		{"n>0, TypeMaxKey", 10, Value{
-			Type: TypeMaxKey}, `{"$maxKey"`},
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("String %s", tc.description), func(t *testing.T) {
+			got := tc.val.String()
+			assert.Equal(t, tc.want, got)
+		})
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.description, func(t *testing.T) {
-			got := tc.val.StringN(tc.n)
+		for n := -1; n <= len(tc.want)+1; n++ {
+			t.Run(fmt.Sprintf("StringN %s n==%d", tc.description, n), func(t *testing.T) {
```

Comment:
Optional: Consider factoring these into two test functions `TestValue_String` and `TestValue_StringN` and moving the test cases to a package-level variable (e.g. `var valueStringTestCases = ...`).